query/src/servers/http/middleware.rs

```diff
@@ -57,21 +64,21 @@ fn get_credential(headers: &HeaderMap) -> Result<Option<Credential>> {
                 let c = Credential::Password {
                     name,
                     password,
-                    hostname: None,
+                    hostname: client_ip,
```

Comment:
can u1@<some_ip> fallback  to  "u1"@"%"` too?

Reply:
do we always use the IP/hostname as a part of  user_id?

or

just may localhost special? and let other addrs map to "%" directly?

Reply:
> can u1@<some_ip> fallback to "u1"@"%"` too?

Yes. see https://github.com/datafuselabs/databend/blob/main/query/src/users/user_mgr.rs#L69-L75